### PR TITLE
Various fixes to make openapi spec valid

### PR DESCRIPTION
**Pull request by louis-lau**
- Paths in openapi don't use colon syntax for params
- Required can't be an empty array
- Required isn't a property in schema items, only parameters. It is already set as a seperate array of keys as per the spec
- Noticed some instances of type: 'alternatives' etc in path and query params, reused the existing joi parsing function to correctly set these. Also sets example for path/query params
- JSON output is now pretty, which should make it easier to read the spec and see what has changed in diffs

### CODE REVIEW

File: restifyOpenapiGenerator.js [hunk 1]
```diff
@@ -157,7 +159,6 @@ class RestifyApiGenerate {
             this.parseJoiObject(null, elems[0], data);
         } else {
             const openApiType = joiTypeToOpenApiTypeMap[joiObject.type]; // even if type is object here then ignore and do not go recursive
-            const isRequired = joiObject._flags.presence === 'required';
```
Comment (NickOvt) [inline restifyOpenapiGenerator.js:160]: Can you explain the reason behind this change?

**Comment by NickOvt** [comment]
Thank you for your changes. They indeed fix some key things that I have missed. Will merge soon. LGTM!

File: restifyOpenapiGenerator.js (hunk 1 above)
Comment (louis-lau) [reply restifyOpenapiGenerator.js:160]: It was only used in the line I removed a little below that, so I cleaned it up :)

Which was removed for this:
> Required isn't a property in schema items, only parameters. It is already set as a seperate array of keys as per the spec

Comment (NickOvt) [reply restifyOpenapiGenerator.js:160]: I see. Thanks! :D

**Review by NickOvt — COMMENTED** [review]
LGTM!

**Review by NickOvt — APPROVED** [review]
.

**Pull request merged**